src/msg/async/AsyncConnection.cc

```diff
@@ -967,17 +967,31 @@ int AsyncConnection::_process_connection()
           ::close(sd);
         }
 
-        sd = net.connect(get_peer_addr());
+        sd = net.nonblock_connect(get_peer_addr());
         if (sd < 0) {
           goto fail;
         }
-        r = net.set_nonblock(sd);
+        center->create_file_event(sd, EVENT_READABLE, read_handler);
+        state = STATE_CONNECTING_RE;
+        break;
+      }
+
+    case STATE_CONNECTING_RE:
+      {
+        if (reconnect == false) {
```

Comment:
There are many ways to test whether it is connected or not, when being wakened. But I think reconnect may be the simplest. In this way, we must break out from the process(), when create file event. Then we should have a situation that prev_state == state, and I make it in reconnect control.